sympy/solvers/solveset.py

```diff
@@ -203,7 +203,7 @@ def invert_real(f_x, y, x):
 def _invert_real(f, g_ys, symbol):
     """Helper function for _invert."""
 
-    if f == symbol:
+    if f == symbol or g_ys is S.EmptySet:
         return (f, g_ys)
```

Comment:
```suggestion
        return (symbol, g_ys)
```
(Same comment/suggestion below.)